### DISCUSSION TITLE
Page Action custom content form submission

### DISCUSSION THREAD

**Discussion by topegret** [filamentphp/filament#5047] [Help]
https://filamentphp.com/docs/2.x/admin/pages/actions#custom-content

in  this document "You may define custom content to be rendered inside your modal, which you can specify by passing a Blade view into the modalContent() method:"  

In my current application development,  I need to submit a custom form in the blade from this modelContent. This form data is  belongs to other model. I don't know how to get the form data when the model "submit" button is been pressed.  Here is my sample code: 
```
<?php
namespace App\Filament\Pages;
use Filament\Pages\Page;
... 
use Filament\Pages\Actions\ActionGroup;
use Filament\Pages\Actions\Action;
....
 protected function getActions(): array
{
    return [
            ActionGroup::make([
              ...
                       Action::make('Edit payment details')
                                    ->action(function (array $data): void {
                                          dd($data);  //try to obtain form data in the modal. 
                                                         })
                            ->modalContent(view('filament.pages.custom.edit_payment')->withPayment($this->profile))
                ...
        ]),
  ...
}
```
So, for in above snippet, how the form inputs in the modal could be fill the $data array?  thanks!

**Comment by danharrin** [comment]
You should use the form builder with `->form()` to get data into the array. You can still specify custom modal content that will be rendered above or below the form.
